src/test/java/org/jenkinsci/plugins/workflow/steps/CoreStepTest.java

```diff
@@ -154,4 +168,36 @@
         st.assertRoundTrip(new CoreStep(aa), "archiveArtifacts 'some-artifacts'");
     }
 
+    public static class BuilderWithEnvironment extends Builder implements SimpleBuildStep {
+
+        @DataBoundConstructor
+        public BuilderWithEnvironment() {
+        }
+
+        @Override
+        public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull EnvVars env, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+            assertNull(env.get("BUILD_ID"));
```

Comment:
How is this `null` rather than `""`?

Reply:
Well, the docs for withEnv say that "FOO=" removes it from the environment, so I assumed it would be null. The test seems to agree :)

Reply:
Huh, https://github.com/jenkinsci/pipeline-plugin/commit/5b7d046506f0f52480328e218e906583792b4bae five years ago :man_shrugging: